### ISSUE TITLE
[Bug] Edge case, Kaizo Crystal

### ISSUE THREAD

**Issue by jhiggins2150** [Ajarmar/universal-pokemon-randomizer-zx#160]
**Describe the bug**

When trying to run the randomizer on an Crystal version that is Kaizo (patched with Lunar IPS) the tool errors out at the ‘getTrainers()’ method. This is likely due to the fact the creator of this patch did heavy modification to the trainers.

Ive running the randomizer with ‘Unchanged’ ticked for trainers and even no changes at all and this error persists.

**To Reproduce**

Patch a Crystal ROM (legally obtained) with the Kaizo patch from SinisterHoodFigure. Attempt to randomize the ROM.

**Expected behavior**

Successful randomization of items both held and found, wild encounters, starters 

**Screenshots**

**System (please complete the following information):**
 - Operating System: 
 - Java Version:
 - Randomizer Version: 

Win 10, Latest, Latest

**Additional context**

I had the thought to try running the randomizer before patching the ROM but this will not work as intended as the patch will overwrite most if not all of the randomization.

tl;dr — thinking that a potential fix could be to comment out the body of the ‘getTrainers()’ method so that trainers are left untouched. 

Unsure what IDE/configuration is needed to build the source code on my own. Have sourced some helpful info that IntelliJ and JetBrains could work! Thanks for reading. And lastly I realize this is an edge case so likely not a real cause for ‘issue’ but that said being able to leave categories completely untouched, such as Trainers in my case, would likely be helpful to others as well. Thank you!

**Comment by jhiggins2150** [comment]
So got IntelliJ installed and am now able to build the project - that's the good news. Bad news is the randomizer still fails to run successfully after commenting out getTrainers. The error shows getEncounters() specifically the bug catching contest. Commenting that out results in a different error. I'm thinking it has to do with offsets? Is this something one can extract from a specific ROM such as Crystal Kaizo, obtain from the creator of the patch, or is this just DOA SOL?

**Comment by Ajarmar** [comment]
We don't support ROM hacks in the randomizer. It's hard to say what exactly is causing the problems here but if data is moved around in the ROM hack then it will cause issues. I agree with your point though - some things (such as trainers) are loaded unnecessarily even if they are unmodified by the randomizer, which we might fix in a future release. But even if the unnecessary loading of trainers is removed, there's no guarantee that there won't be conflicts with other settings. (There are also some things that _do_ need to be loaded even if they are unmodified, such as the list of Pokemon.)

**Comment by jhiggins2150** [comment]
> We don't support ROM hacks in the randomizer. It's hard to say what exactly is causing the problems here but if data is moved around in the ROM hack then it will cause issues. I agree with your point though - some things (such as trainers) are loaded unnecessarily even if they are unmodified by the randomizer, which we might fix in a future release. But even if the unnecessary loading of trainers is removed, there's no guarantee that there won't be conflicts with other settings. (There are also some things that _do_ need to be loaded even if they are unmodified, such as the list of Pokemon.)

So I'm reaching here and it's a lot to ask but any thoughts or ideas on how I can continue pursuing this? I would love to make it happen and if I can get it working I'm happy to share it with everyone. Not sure exactly what I'm asking for... how to locate offsets for trainers, items, encounters, etc? I don't presume to know anything about this but I have to think there is a way to locate offsets and update them for a given hack, such as Kaizo.

**Comment by Ajarmar** [comment]
I'd be hesitant to accept contributions that add support for ROM hacks, primarily because of the fact that they may have to be maintained in the future if we change something about how we want the offsets to be structured (or make some other change that would break compatibility). That said, if you just want to have it on your own fork, go right ahead.

To answer your question though: I'm not sure there's an easy way of doing this. If you know what the data you're looking for looks like then maybe you could find it by opening the ROM with a hex editor and searching for the data, otherwise I think you'd have to use an emulator with a debugger to track it down.
